docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
What we may want to change about the interface is that we pass in the KDF class, not the instance, and let agree() instantiate it internally every time it is called.
As is exchange+ agree can be called only once indeed because then the KDC will return an error as it already derived a key.